core/utils/misc.hpp

```diff
@@ -33,57 +32,37 @@ namespace iresearch {
 
 // Convenient helper for simulating 'try/catch/finally' semantic
 template<typename Func>
-class finally {
+class [[nodiscard]] Finally {
  public:
   static_assert(std::is_nothrow_invocable_v<Func>);
 
-  explicit finally(Func&& func) : func_{std::forward<Func>(func)} {}
-  ~finally() { func_(); }
+  // If you need some of it, please use absl::MakeCleanup
```

Comment:
Why not completely move to absl::Cleanup ?

Reply:
We discussed it earlier with @gnusi and for simple expression we want simple (without flag, placement new, etc) abstraction.

But copy/move finally is unsafe (func will be called twice or more time)